v5_MIGRATION_GUIDE.md

```diff
@@ -493,4 +494,36 @@ class MyActivity : FragmentActivity() {
 -   override fun onSEPADirectDebitFailure(error: java.lang.Exception) {
 -        // handle error
 -   }
-}
\ No newline at end of file
+}
+```
+
+## Visa Checkout
+
+The Visa Checkout integration has been updated to improve result handling.
+
+```diff
+class MyActivity : FragmentActivity() {
+
+-   private lateinit var braintreeClient: BraintreeClient
+    private lateinit var visaCheckoutClient: VisaCheckoutClient
+
+    fun initializeClients() {
+-       braintreeClient = BraintreeClient(context, "TOKENIZATION_KEY_OR_CLIENT_TOKEN")
+-       visaCheckoutClient = VisaCheckoutClient(braintreeClient)
++       visaCheckoutClient = visaCheckoutClient(context, "TOKENIZATION_KEY_OR_CLIENT_TOKEN")
+    }
+
+    fun onPaymentButtonClick() {
+-       visaCheckoutClient.tokenize(request)
++       sepaDirectDebitClient.tokenize(request) { visaCheckoutResult ->
++           when(visaCheckoutResult) {
++               is (VisaCheckoutResult.Failure) -> { 
++                   // handle visaCheckoutResult.error
++               is (VisaCheckoutResult.Success) -> {      
++                   // handle visaCheckoutResult.nonce
++               }
```

Comment:
```suggestion
+               is VisaCheckoutResult.Failure -> { 
+                   // handle visaCheckoutResult.error
+               is VisaCheckoutResult.Success -> {      
+                   // handle visaCheckoutResult.nonce
+               }
```
nit: I noticed this recently. It's the one part of Kotlin that feels more like Swift.

Reply:
Oh didn't know you could do that - very cool I'll update the rest in anther PR